### DISCUSSION TITLE
One is the enemy

### DISCUSSION THREAD

**Discussion by jh62** [gamerevar/proyecto-comunitario-1-gd#53] [Archivados]
_Puse el título en inglés porque es más atractivo_ 🤣

# Título

**Uno es el enemigo**

## Sinopsis

En un **mismo planeta** se enfrentan 2 grupos: uno es el habitante y el otro el _"invasor"_. Pero, ¿**quién es el enemigo**? 
El grupo invasor está buscando una escapatoria y el otro, el nativo, está defendiendo su lugar, su mundo de algo desconocido.

## Mecánicas

- Combate por turnos al estilo X-COM.
- Las _vidas_ del jugador son la cantidad de unidades que le queden en su equipo.
- Las unidades mueren **permanentemente** y no se reponen (salvo excepciones, como rescatar a alguien).
- El personaje principal da moral a todo el resto. Si muere, el juego no termina, pero continuar lo hará extremadamente difícil de ganar ya que el resto perderá una gran parte de moral, la cual afecta al rendimiento íntegro de cada unidad.

## Cómo se adapta a la premisa

Un juego puede basarse sobre los nativos (con esta palabra no me refiero a indios o cavernícolas, puede ser cualquier setting) y el otro sobre la fuerza invasora. El primero puede funcionar como un combate por turnos al estilo X-COM:

![](https://jasonlefkowitz.net/wp-content/uploads/2012/10/xcom-1994.jpg)

El otro igual o tal vez como un _vertical scrolling shooter_:

![](https://academickids.com/encyclopedia/images/9/90/Raiden_%28game%29.png)

## Historia

La historia en ambos se puede contar a través de cutscenes piolas y ambas expondrán el punto de vista de cada bando, justificando sus acciones y quedando a criterio del jugador quién es el bueno y quién es el malo.

**Comment by LeandroCuello** [comment]
me gusta mucho la idea de hacer un combate por turnos al estilo X-COM! y la idea de la historia me parece buena.

**Comment by doctorhibert** [comment]
Cuán dificil sería programar la inteligencia artificial para un juego como XCOM?

**Reply by jh62** [reply]
Si la IA es simple, diría que moderado. Yo creo que con una sólida state machine podés lograr cosas interesantes. La verdad es que hasta que no lo intentemos, no sabemos. En en peor de los casos los npc se quedan parados y se limitan a defenderse/atacar en su rango de visión.

**Reply by LeandroCuello** [reply]
También va a depender de las mecánicas del juego. Va a ver ventaja por terreno alto? Van a tener variedad de armas? El terreno cambia? Entre más complejo sea más difícil va a ser si hacemos un táctico tenemos que tener cuidado con cada mecánica que queramos agregar.

**Reply by emrodk** [reply]
Cómo todo lleva su tiempo pero no debería ser problemático, al ser por turnos y un terreno fijo, que bien podría estar basado en tiles, te ahorras programar cosas cómo que los enemigos sean capaces de esquivar obstáculos, algo vital en un FPS por ejemplo.

**Reply by jh62** [reply]
> También va a depender de las mecánicas del juego. Va a ver ventaja por terreno alto? Van a tener variedad de armas? El terreno cambia? Entre más complejo sea más difícil va a ser si hacemos un táctico tenemos que tener cuidado con cada mecánica que queramos agregar.

No lo complicaría con elevaciones, pero si es por tipo de tile, seria muy trivial implementar algo.

**Reply by doctorhibert** [reply]
Bueno, se podría hacer como minimo una IA simple, y lo bueno es que esta idea puede ir de la mano de las otras ideas que postuló la gente
